### PR TITLE
[MAINT] make `__all__` and API doc more consistent

### PR DESCRIPTION
**Pull request by Remi-Gau**
- relates to #4137

Modules should explicitly declare the names in their public API using the `__all__` attribute but that we have some things in the `__all__` that is not in the API doc.

This PR fixes some issues:
- removes functions that are in the public API of some modules but not in the doc
- orders `__all__` alphabetically to make this comparison easier in the future

### CODE REVIEW

File: nilearn/glm/first_level/__init__.py [hunk 1]
```diff
@@ -2,7 +2,6 @@
     check_design_matrix,
     make_first_level_design_matrix,
 )
-from nilearn.glm.first_level.experimental_paradigm import check_events
```
Comment (Remi-Gau) [inline nilearn/glm/first_level/__init__.py:5]: check_events not in API doc

File: nilearn/image/__init__.py [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 Like, for example, a (3+)D block of data, and an affine.
 """
+# TODO move copy_img out of nilearn._utils..niimg
 from .._utils.niimg import copy_img
+
+# TODO move concat_niimgs out of nilearn._utils.niimg_conversions
 from .._utils.niimg_conversions import concat_niimgs as concat_imgs
```
Comment (Remi-Gau) [inline nilearn/image/__init__.py:9]: Tempted to say that things that are part of the user facing API should NOT be in a part of the package that is private.

File: nilearn/maskers/__init__.py [hunk 3]
```diff
@@ -1,6 +1,5 @@
 """The :mod:`nilearn.maskers` contains masker objects."""
 
-from ._utils import compute_middle_image
```
Comment (Remi-Gau) [inline nilearn/maskers/__init__.py:3]: compute_middle_image not in the API doc

File: nilearn/plotting/__init__.py [hunk 4]
```diff
@@ -93,35 +93,35 @@ def _set_mpl_backend():
 )
 
 __all__ = [
```
Comment (Remi-Gau) [inline nilearn/plotting/__init__.py:95]: Probably want to make sure the __all__ are sorted alphabetically to make easier to cross check things

File: nilearn/reporting/__init__.py [hunk 5]
```diff
@@ -5,6 +5,7 @@
 Author: Martin Perez-Guevara, Elvis Dohmatob, 2017
 """
 
+# TODO consider making _get_clusters_table module public?
 from nilearn.reporting._get_clusters_table import get_clusters_table
```
Comment (Remi-Gau) [inline nilearn/reporting/__init__.py:9]: User facing function comes from a private module

File: nilearn/masking.py [hunk 6]
```diff
@@ -19,12 +19,24 @@
...
+    "compute_epi_mask",
+    "compute_multi_background_mask",
+    "compute_multi_brain_mask",
+    "compute_multi_epi_mask",
+    "intersect_masks",
+    "unmask",
+]
+
+
+class _MaskWarning(UserWarning):
```
Comment (Remi-Gau) [inline nilearn/masking.py:35]: Warning not used outside of this module

File: nilearn/glm/__init__.py [hunk 7]
```diff
@@ -24,22 +24,20 @@
 )
 
 __all__ = [
-    "first_level",
-    "second_level",
```
Comment (Remi-Gau) [inline nilearn/glm/__init__.py:28]: note that those were duplicated.

**Review by ymzayek — APPROVED** [review]

**Review by ymzayek — COMMENTED** [review]
I think you just need to adjust the number in test_number_public_classes

**Comment by Remi-Gau** [comment]
> I think you just need to adjust the number in test_number_public_classes

thanks!!! 🤗 

should be fixed now

**Review by bthirion — APPROVED** [review]
LGTM, thx

**Pull request merged**